### PR TITLE
Configure: Make --strict-warnings meaningful with MSVC cl

### PR DESCRIPTION
**Pull request by levitte**
We also add this to our x86 builds on appveyor

### CODE REVIEW

File: appveyor.yml [hunk 1]
```diff
@@ -18,7 +18,7 @@ before_build:
             $env:TARGET="VC-WIN32 no-asm"
         } Else {
             $env:VCVARS_PLATFORM="amd64"
-            $env:TARGET="VC-WIN64A-masm"
+            $env:TARGET="VC-WIN64A-masm --strict-warnings"
```
Comment (mattcaswell) [inline appveyor.yml]: Hmm....I thought 32-bit builds were clean and 64-bit ones are not at the moment? Or am I wrong and its the other way around?

Comment (levitte) [reply appveyor.yml]: You're entirely correct.  Will fix

Comment (mattcaswell) [reply appveyor.yml]: Ah...10-main.conf currently adds "/WX" for VC-WIN32 builds. So probably appveyor should have "--strict-warnings" for VC-WIN32 builds, and then remove that flag from 10-main.conf.

Comment (levitte) [reply appveyor.yml]: Yeah, that's what I ended up doing

**Comment by levitte** [comment]
Appveyor approves, so I'm taking this out of WIP / draft.  Time for final review

**Review by mattcaswell — APPROVED** [review]

**Comment by levitte** [comment]
Merged.

b4a7b4ec4acc712b1f22a83966ac986b510f25d8 Configure: Make --strict-warnings meaningful with MSVC cl

**Comment by levitte** [comment]
Mental note to sself: This needs to be backported to 1.1.1, ref #8074

**Comment by levitte** [comment]
It was *almost* trivial to cherry-pick it.  To be safe, I'm making a separate backport PR

**Comment by levitte** [comment]
Backport in #10365

**Pull request closed without merging**